extensions/pkg/controller/worker/genericactuator/machine_controller_manager.go

```diff
@@ -124,8 +125,58 @@ func (a *genericActuator) waitUntilMachineControllerManagerIsDeleted(ctx context
 }
 
 func (a *genericActuator) scaleMachineControllerManager(ctx context.Context, logger logr.Logger, worker *extensionsv1alpha1.Worker, replicas int32) error {
+	deployment := &appsv1.Deployment{}
+	if err := a.client.Get(ctx, kutil.Key(worker.Namespace, McmDeploymentName), deployment); err != nil {
+		if apierrors.IsNotFound(err) {
+			logger.Info("Machine-controller-manager is not deployed yet")
+			return nil
+		}
+		return err
+	}
 	logger.Info("Scaling machine-controller-manager", "replicas", replicas)
-	return kubernetes.ScaleDeployment(ctx, a.client, kutil.Key(worker.Namespace, McmDeploymentName), replicas)
+	if err := kubernetes.ScaleDeployment(ctx, a.client, kutil.Key(worker.Namespace, McmDeploymentName), replicas); err != nil {
+		return err
+	}
+	return a.waitUntilMachineControllerManagerIsScaled(ctx, worker)
```

Comment:
It would be more clear to move this out of the `scaleMachineControllerManager` function and rather call it in `Restore`.